docs/the-new-architecture/use-app-template.md

```diff
@@ -1,7 +1,134 @@
 ---
 id: use-app-template
-title: Using the New App Template
+title: Creating a New Architecture App
 ---
 
-This section describes the new app template generated by the React Native CLI and how to use it properly.
-It should explain how to create a new app, which command should be issued to install the dependencies and to run it on the different platforms.
+import Tabs from '@theme/Tabs'; import TabItem from '@theme/TabItem'; import constants from '@site/core/TabsConstants';
+
+import RemoveGlobalCLI from '.././\_remove-global-cli.md';
+
+This page will help you create a new React Native app that uses the New Architecture.
+
+## Development environment
+
+Before continuing, make sure you've followed all the steps in [Setting up the
+development environment](getting-started.md), under the **React Native CLI
+Quickstart** tab.
+
+If following the setup guide, stop when you reach the section **Running your React Native
+Application**, and resume following this guide.
+
+**Note**: The New Architecture cannot currently be enabled in an Expo app.
+
+## Creating a new application
+
+<RemoveGlobalCLI />
+
+If you already have your development environment set up, create a new React Native project from the template:
+
+```shell
+npx react-native init AwesomeProject
+```
+
+If specifying a versioned template, use version 0.68 or later.
+
+## Configuration
+
+Follow the steps below to enable the New Architecture and build the app.
+
+### Enable Hermes
+
+Hermes is an open-source JavaScript engine optimized for React Native. [Hermes will be the default engine in the future](https://github.com/reactwg/react-native-new-architecture/discussions/4), and we highly recommend you use it.
+
+Please [follow the instructions on the React Native website](hermes.md) in order to enable Hermes in your application.
+
+### Enable the New Architecture
+
+#### Target OS
+
+<Tabs groupId="platform" defaultValue={constants.defaultPlatform} values={constants.platforms} className="pill-tabs">
+<TabItem value="ios">
+
+Navigate to the `ios` directory and run the following:
+
+```shell
+# from `ios` directory
+RCT_NEW_ARCH_ENABLED=1 bundle exec pod install
+```
+
+Then build and run the app as usual:
+
+```shell
+npx react-native run-ios
+```
+
+> **Tip**: You need to re-run `pod install` each time a dependency with native
+> code changes. Make this command easier to run by adding it to `scripts` to your
+> project's `package.json` file:
+>
+> ```
+> "scripts": {
+>   "pod-install": "RCT_NEW_ARCH_ENABLED=1 bundle exec pod install"
+> }
+> ```
+>
+> and run it with `yarn pod-install`.
+
+#### Troubleshooting
+
+##### `react-native run-ios` fails
+
+If you see a build failure from `react-native run-ios`, there may be cached files from a previous build with
+the old architecture. Clean the build cache and try again:
+
+1. Open the project `ios/project.xcworkspace` in Xcode
+2. In XCode, choose Product > Clean Build Folder
+3. Try building and running the app again
+
```

Comment:
Does it make sense to explain how to clean up the pods as well? Something like:

```suggestion
4. If the error persists, go to the `ios` folder and run: `rm -rf Pods Podfile.lock build`. This cleans up the pods and the build folder who may be stale. Notice that these instruction requires another execution of `yarn pod-install`. 
```

Reply:
Yup, good call.

Reply:
I opted for always recommending removing the Pods; as general troubleshooting advice, this seems reasonable to me?

Reply:
Also, because of the inclusion of using `yarn pod-install` again, I opted to swap the `npx react-native` calls for their corresponding `yarn` scripts in the template, it seems good to match that where possible, and the `yarn` invocations are much more concise.

Reply:
> I'm wondering whether we should also explain that the build will take longer after deleting the pods, but I guess it is implied, given that we are purging the cache

Yeah. I also think that for any user who would not expect this, they probably also don't have expectations about build time (i.e., they are new to the ecosystem and trust the tooling). And over time I think this is a pattern that becomes apparent to RN developers, even if they don't know the iOS ecosystem.